service/bfg/bfg.go

```diff
@@ -1474,6 +1477,30 @@ func (s *Server) refreshCacheAndNotifiyL2Keystones(pctx context.Context) {
 	go s.handleL2KeystonesNotification()
 }
 
+// to prevent against keystones being mined too far in the future, determine
+// a cutoff time 10 minutes in the future.  calculate the l2 block height at
+// that time, don't include keystones higher than that
+func (s *Server) l2KeystoneIgnoreAfter() int64 {
+	log.Tracef("refreshL2KeystoneIgnoreAfter")
+	defer log.Tracef("refreshL2KeystoneIgnoreAfter exit")
+
+	// set cut off 10 minutes in the future
+	// calculate the number of expected blocks between then and our baseline
+	// block
+
+	cutoff := time.Now().Add(10 * time.Minute).Unix()
+	expectedTimeElapsedInBlocks := (cutoff - s.cfg.BaselineL2BlockTimestamp) / hemi.L2BlockTimeSeconds
+	expectedHighestBlock := s.cfg.BaselineL2BlockHeight + expectedTimeElapsedInBlocks
+
+	log.Tracef("the time is cutt off at %d, we expected %d to be the highest block", cutoff, expectedHighestBlock)
```

Comment:
I think you've got too many letters in `cut`

Reply:
ah yep...fixing